tcms/testplans/tests/test_edit_plan.py

```diff
@@ -0,0 +1,58 @@
+from http import HTTPStatus
+
+from django.urls import reverse
+
+from tcms.tests import user_should_have_perm, BasePlanCase
+
+
+class EditPlanViewTest(BasePlanCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.testplan_edit_url = reverse('plan-edit', args=[cls.plan.pk])
+
+        cls.testplan_edit_data = {
+            'author': cls.tester.pk,
+            'product': cls.plan.product.pk,
+            'product_version': cls.plan.product_version.pk,
+            'type': cls.plan.type.pk,
+            'name': cls.plan.name,
+
+            'email_settings-0-auto_to_plan_author': 'on',
+            'email_settings-0-auto_to_case_owner': 'on',
+            'email_settings-0-auto_to_case_default_tester': 'on',
+            'email_settings-0-notify_on_case_update': 'on',
+            'email_settings-0-notify_on_plan_update': 'on',
+
+            'email_settings-0-plan': '',
+            'email_settings-0-id': cls.plan.emailing.pk,
+            'email_settings-TOTAL_FORMS': '1',
+            'email_settings-INITIAL_FORMS': '1',
+            'email_settings-MIN_NUM_FORMS': '0',
+            'email_settings-MAX_NUM_FORMS': '1',
+        }
+
+        user_should_have_perm(cls.tester, perm='testplans.change_testplan')
+
+    def test_404_if_testplan_id_not_exist(self):
+        url = reverse('plan-edit', args=[99999])
+        response = self.client.get(url)
+        self.assertEqual(HTTPStatus.NOT_FOUND, response.status_code)
+
+    def test_show_testplan_edit_page(self):
+        response = self.client.get(self.testplan_edit_url)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertContains(response, self.plan.name)
```

Comment:
You can assert a bit more things here. Like for example are we on the Edit Test plan page indeed! (see page title)

Reply:
Also the `notify_formset` in `context` is something that is important and could break so an assert that all of the values are as expected will verify everything is fine in that part of the code.